amcl/src/amcl_node.cpp

```diff
@@ -1572,57 +1578,104 @@ AmclNode::laserReceived(const sensor_msgs::LaserScanConstPtr& laser_scan)
          }
        */
 
-
-      // odom_correction
+      // odom correction
       if (!odom_correction_)
       {
         pose_pub_.publish(p);
         last_published_pose = p;
       }
+      else if (pause_odom_correction_)
```

Comment:
初期化直後からinitial_poseまでのみ有効な分岐になっている。
```c++
if (!odom_correction_)
```
より先にチェックするか、else ifではなく、ifとして独立させた方が良いのでは?

わかりやすさ、pause_odom_correctionがtrueで、odom_correction_がfalseのエッジケースが存在しないかを懸念。
その時は、この条件式に入っていかない。

The branch is only valid from immediately after initialization to initial_pose.
```c++
if (!odom_correction_)
```
Shouldn't it be checked before or stand alone as an if instead of an else if?

Concerned about clarity and whether there are edge cases where pause_odom_correction is true and odom_correction_ is false.
If so, they do not go into this conditional expression.

Reply:
fixed in 3df40539b9d00e3612ffeb3726cc5ee295daeee5